### PR TITLE
chore: porting of benchmark test for check with new model

### PR DESCRIPTION
**Pull request by adriantam**
## Description
1. Porting of benchmark test to use refactored check
2. Remove old check code and its associated test

## References
Close https://github.com/openfga/openfga/issues/542

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/server/test/list_objects.go [hunk 1]
```diff
@@ -27,7 +27,9 @@ const (
 	defaultListObjectsDeadline   = 5 * time.Second
 	defaultListObjectsMaxResults = 5
 
-	checkConcurrencyLimit = 100
+	checkConcurrencyLimit   = 100
+	defaultResolveNodeLimit = 25
```
Comment (adriantam) [inline pkg/server/test/list_objects.go:31]: these const. were in the test/check.go and used by this file and other test file.

File: tests/tests.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func StartServer(t *testing.T, cfg *run.Config) context.CancelFunc {
+func StartServer(t testing.TB, cfg *run.Config) context.CancelFunc {
```
Comment (jon-whit) [inline tests/tests.go:12]:
```suggestion
func StartServer(t *testing.TB, cfg *run.Config) context.CancelFunc {
```

Pointer to

File: pkg/server/test/list_objects.go (hunk 1 above)
Comment (jon-whit) [reply pkg/server/test/list_objects.go:31]: Let's just add these in `pkg/server/test/server.go` instead. They can/may be used across test files.

File: tests/check/check_test.go [hunk 3]
```diff
@@ -100,3 +106,197 @@ func testBadAuthModelID(t *testing.T, engine string) {
...
+	resp, err := client.CreateStore(ctx, &pb.CreateStoreRequest{Name: "check benchmark without trace"})
+	require.NoError(b, err)
+
+	storeID := resp.GetId()
+	_, err = client.WriteAuthorizationModel(ctx, &pb.WriteAuthorizationModelRequest{
+		StoreId:         storeID,
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+		TypeDefinitions: parser.MustParse(githubModel),
+	})
+	require.NoError(b, err)
```
Comment (jon-whit) [inline tests/check/check_test.go:176]: Newline please.

File: tests/check/check_test.go [hunk 4]
```diff
@@ -100,3 +106,197 @@ func testBadAuthModelID(t *testing.T, engine string) {
...
+
+	cancel := tests.StartServer(b, cfg)
+	defer cancel()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr,
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(b, err)
+	defer conn.Close()
```
Comment (jon-whit) [inline tests/check/check_test.go:163]: Newline please

File: tests/check/check_test.go [hunk 5]
```diff
@@ -100,3 +106,197 @@ func testBadAuthModelID(t *testing.T, engine string) {
...
+	client := pb.NewOpenFGAServiceClient(conn)
```
Comment (jon-whit) [inline tests/check/check_test.go:207]: L194-L207 is all boilerplate code that is used in each `benchmarkCheck...` benchmark function. Can we consolidate please?

**Review by jon-whit — CHANGES_REQUESTED** [review]
Left some feedback

File: tests/tests.go (hunk 2 above)
Comment (adriantam) [reply tests/tests.go:12]: testing.TB itself is an interface while testing.T is the struct.  This is why you need pointer for testing.T while you don't need / cannot have pointer for TB

File: pkg/server/test/list_objects.go (hunk 1 above)
Comment (adriantam) [reply pkg/server/test/list_objects.go]: Done

File: tests/check/check_test.go (hunk 5 above)
Comment (adriantam) [reply tests/check/check_test.go]: done

File: tests/tests.go (hunk 2 above)
Comment (jon-whit) [reply tests/tests.go:12]: Oops, my bad. 👍

**Review by jon-whit — APPROVED** [review]

**Review by craigpastro — APPROVED** [review]
LGTM. I think there is the `contextualtuples` package. Are those files still being used?

**Comment by adriantam** [comment]
> LGTM. I think there is the `contextualtuples` package. Are those files still being used?

Yes.  See https://github.com/openfga/openfga/blob/6e0dd8b41806877bfbe0965e00e79a37b58decfa/pkg/server/commands/list_objects.go#L10

**Pull request merged**